docker/validator-dynamic/build-aws.sh

```diff
@@ -0,0 +1,67 @@
+#!/bin/bash
+# Copyright (c) The Libra Core Contributors
+# SPDX-License-Identifier: Apache-2.0
+# build-aws.sh builds the validator-dynamic image on AWS Codebuild and pushes the image to AWS ECR.
+# Examples:
+# build-aws.sh --version pull/123 # Builds the image from pull request 123 and tags it with dev_pr_123
+# build-aws.sh --version u29a020 # Builds the image from commit u29a020 and tags it with dev_u29a020
+
+if ! which jq &>/dev/null; then
+  echo "jq is not installed. Please install jq"
+  exit 1
+fi
+
+function print_usage() {
+  echo "Usage: build-aws.sh --version pull/123"
+  exit 1
+}
+
+while [[ "$1" =~ ^- ]]; do case $1 in
+  --version )
+    shift;
+    if [[ "$1" =~ ^pull ]]; then
+      SOURCE_VERSION="refs/${1}/head"
+      TAGS="dev_pr_${1/\//_}"
```

Comment:
Something to consider: if the PR has updates, the prior image would become untagged. ECR would look littered.

Reply:
Should we do smth like `$User_$ShortGitHash` instead as a tag?

Reply:
I have updated it such that every build also gets tagged with `dev_${ShortGitHash}`. This should be always unique and with this we don't need the `$User` prefix.  This should also avoid the ECR littering issue.

I have also added a way to add custom tags with `--tags` which can be used to add arbitrary user tags to every build.